### PR TITLE
Peta User Widget

### PR DESCRIPTION
**Pull request by Tiaraha-funny**
Hello team,

The peta-user-widget branch is created to handle the hovering the account link on the navigation on top after logging in. When that person hovers the account link, he/she can see a popup modal that displays a clickable profile photo and a username to go to my account page, a clickable offer and a number of that person's offer/offers that leads to my offer page and a log out link that allows the user to reset all the page then log in again if he/she wants to do something on his/her account.

Please have a look at it and any feedback will highly appreciated.

ClickUp: https://app.clickup.com/t/1w9nkva
Refs: #1w9nkva
Figma: https://www.figma.com/file/Ffm0kRm77Hz0khvZNGh4Oc/Onja-Big-Project?node-id=3418%3A3908

Thanks in advance for the suggestions.

Peta.

### CODE REVIEW

File: src/pages/UserWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import { fonts } from "../globalStyles/fonts";
+
+export const MyAccountWidget: FC<ModalProps> = ({ hide }) => {
+  const user = useSelector(userSelector);
+  const offers = useSelector(adsSelector);
+  const dispatch = useDispatch();
+
+  const countOffers =
+    offers &&
+    offers.filter((el) => (el.username === user.username ? el.username : 0));
```
Comment (andrzej-onja) [inline src/pages/UserWidget.tsx:19]: you should have used API
`query{getUserAds(userId: "61af3e2a83117000298ae3d6"){id,title,body,category{title}, user{username, email}}}`

File: src/pages/UserWidget.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            </div>
+          </Link>
+
+          <Link to={Paths.MY_OFFER} key={user.createdAt}>
+            Offers: <OfferNum>{countOffers.length}</OfferNum>
+          </Link>
+        </ProfileWrapper>
+
+        <span onClick={hide}>
+          <LogoutButton onClick={fetchUser}>
```
Comment (andrzej-onja) [inline src/pages/UserWidget.tsx:47]: fetchUser ???  this name is misleading should be logoutUser or something

File: src/pages/UserWidget.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+import { useDispatch, useSelector } from "react-redux";
+import { logout, userSelector } from "../redux/slices/userSlice";
+import { Paths } from "../paths";
+import { adsSelector, fetchAds } from "../redux/slices/adsSlice";
+import profile from "../images/adolfia.jpg";
+import { ModalProps } from "../components/ModalWidget/useModal";
+import { fonts } from "../globalStyles/fonts";
+
+export const MyAccountWidget: FC<ModalProps> = ({ hide }) => {
+  const user = useSelector(userSelector);
```
Comment (andrzej-onja) [inline src/pages/UserWidget.tsx:13]: Correct use of selector

File: src/pages/UserWidget.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
+import { FC, useEffect } from "react";
+import { Link } from "react-router-dom";
+import styled from "styled-components";
+import { useDispatch, useSelector } from "react-redux";
+import { logout, userSelector } from "../redux/slices/userSlice";
+import { Paths } from "../paths";
+import { adsSelector, fetchAds } from "../redux/slices/adsSlice";
+import profile from "../images/adolfia.jpg";
```
Comment (andrzej-onja) [inline src/pages/UserWidget.tsx:8]: This should be a dynamic, photos from user data

File: src/pages/MyOffersPage.tsx [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    <WrapperOffer>
+      <MyOffer offers={offers} username={"username"} id={"123412132"} />
+    </WrapperOffer>
+  );
+}
+
+const MyOffer: React.FC<OffersListProps> = ({ offers }) => {
+
+    const user = useSelector(userSelector);
+    const myOffer = offers && offers.filter(offer => offer.username === user.username ? offer.username : '')
```
Comment (andrzej-onja) [inline src/pages/MyOffersPage.tsx:49]: you should have used API
query{getUserAds(userId: "61af3e2a83117000298ae3d6"){id,title,body,category{title}, user{username, email}}}

File: src/pages/MyOffersPage.tsx [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+  firstName: string;
+  lastName: string;
+}
+
+export interface Photo {
+  url: string;
+  info: string;
+  isPrimary: boolean;
+}
+interface OfferData {
```
Comment (ganamavo) [inline src/pages/MyOffersPage.tsx:23]: Looks like a type

File: src/pages/MyOffersPage.tsx [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+import { adsSelector } from "../redux/slices/adsSlice";
+import { useSelector } from "react-redux";
+import { userSelector } from "../redux/slices/userSlice";
+
+interface User {
+  firstName: string;
+  lastName: string;
+}
+
+export interface Photo {
```
Comment (ganamavo) [inline src/pages/MyOffersPage.tsx:18]: I'd use a type instead of an iterface but not a big deal

**Comment by andrzej-onja** [comment]
closing as it was a part of graded project only

**Pull request closed without merging**